packages/netsuite-adapter/src/config/config_creator.ts

```
@@ -77,11 +78,84 @@ const updatedFetchTarget = (
   }
 }
 
+const includeFileCabinetFolders = (config: NetsuiteConfig): string[] =>
+  config.includeFileCabinetFolders?.map(
+    folderName =>
+      `^${folderName.startsWith('/') ? '' : '/'}${folderName}${folderName.endsWith('/') ? '' : '/'}.*`
+  ) ?? []
+
+const includeCustomRecords = (config: NetsuiteConfig): FetchTypeQueryParams[] => {
+  if (config.includeCustomRecords === undefined || config.includeCustomRecords.length === 0) {
+    return []
+  }
+  if (config.includeCustomRecords.includes(INCLUDE_ALL)) {
+    return [{ name: ALL_TYPES_REGEX }]
+  }
+  return [{ name: config.includeCustomRecords.join('|') }]
+}
+
+const excludeInactiveRecords = (config: NetsuiteConfig): CriteriaQuery[] => {
+  if (config.includeInactiveRecords === undefined || config.includeInactiveRecords.includes(INCLUDE_ALL)) {
+    return []
+  }
+
+  const typesToInclude = config.includeInactiveRecords
+    .flatMap(typeName => (typeName === FILE_CABINET ? fileCabinetTypesNames : typeName))
+
+  const inactiveRecordsToExcludeRegex = typesToInclude.length === 0
+    ? ALL_TYPES_REGEX
+    // match all except for the exact strings in typesToInclude
+    : `(?!(${typesToInclude.join('|')})$).*`
+
+  return Object.values(INACTIVE_FIELDS).map(fieldName => ({
+    name: inactiveRecordsToExcludeRegex,
+    criteria: {
+      [fieldName]: true,
+    },
+  }))
+}
+
+const excludeDataFileTypes = (config: NetsuiteConfig): string[] => {
+  if (config.includeDataFileTypes === undefined) {
+    return []
+  }
+  const dataFileTypesToInclude = config.includeDataFileTypes.join('|')
+  const dataFileTypesToExclude = Object.values(DATA_FILE_TYPES)
+    .filter(fileType => !(new RegExp(`\\b${fileType}\\b`, 'i')).test(dataFileTypesToInclude))
```

This implementation is really confusing to me
I think if this is supposed to set the example for future configuration refactors we should rethink this approach.
I had a really hard time figuring out where is the code handles `DATA_FILE_TYPES_GROUPS`, the relation here is very implicit, only after I read the code and understood the details I realized that this is actually the thing that makes it work.
if this was not a PR that had a small number of changes, I don't know if I would have been able to find this...

I also think that this is too much complicated logic for a configuration field.
it feels strange that a configuration field basically accepts anything as a valid value and silently ignores most of the words there except for a few specific words which are not really documented anywhere

I am not saying this won't technically work, but I really really think this is not something we would want to maintain and replicate to other use cases

---

What are the alternatives for this field? support only `DATA_FILE_TYPES_GROUPS`? support `DATA_FILE_TYPES_GROUPS` and individual data types from the list? what should be under the restriction annotation?

---

personally I would not implement groups for this, I think users can configure individual suffixes, I really think that would be more convenient and clear.

if you strongly think the groups are important here, then there are a couple of options:
1. only groups are allowed and when you use the configuration you behave according to the group names, the configuration itself only knows about group names and has no knowledge of suffixes at all. in this case the restriction should only allow group names.
2. both groups and suffixes are allowed, this requires a bit of code similar to the code you wrote - if you see a group, you need to replace it with the suffixes it represents (without parsing the group name, the groups and their meaning are hard coded). in this case the restriction must contain all the groups and all the suffixes

I think option 2 is still on the border of being overly complicated for this use case...

---

I think we can start with option 1 and maybe change it in the future.

---

@dantal4 why do we stick with the groups here? due to the product requirement? I think we can reconsider it and remove that requirement. Of course after verifying with Ofer & Miki.